zpay32/invoice_test.go

```diff
@@ -727,6 +756,20 @@ func TestNewInvoice(t *testing.T) {
 			valid:          true,
 			encodedInvoice: "lnbc241pveeq09pp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqdqqnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66jd3m5klcwhq68vdsmx2rjgxeay5v0tkt2v5sjaky4eqahe4fx3k9sqavvce3capfuwv8rvjng57jrtfajn5dkpqv8yelsewtljwmmycq62k443",
 		},
+		{
+			// Invoice with payment address.
+			newInvoice: func() (*Invoice, error) {
+				return NewInvoice(&chaincfg.MainNetParams,
+					testPaymentHash, time.Unix(1503429093, 0),
+					Amount(testMillisat24BTC),
+					Description(testEmptyString),
+					Destination(testPubKey),
+					PaymentAddr(testPaymentAddr),
+				)
+			},
+			valid:          true,
+			encodedInvoice: "lnbc241pveeq09pp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqdqqnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66sp5qszsvpcgpyqsyps8pqysqqgzqvyqjqqpqgpsgpgqqypqxpq9qcrsmmw6t0kplachqfzd76tvuuln6k2zqmxf2u57xcla9lcxfcrsrd48q5ekp2xul50yc6x4wu5qaen3zm3wvuukf9fespkytwqzgya5cxgpeyed0p",
+		},
```

Comment:
Don't you want to use the same invoice as in the payment addr test above? So the encoded invoice is exactly the same and we can be more certain that it works?

Reply:
The first test checks both encoding and decoding. This test checks the functional options I guess.

Reply:
looking at it now, the entire second test suite looks redundant.

Reply:
removed the second test, since it tests both encode and decode